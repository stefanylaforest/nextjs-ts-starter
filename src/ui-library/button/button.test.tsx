import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faShoppingCart } from "@fortawesome/free-solid-svg-icons";

import Button from "./button";

afterAll(cleanup);

describe("Icon Only Button", (): void => {
  it("It should have an aria-label attribute ", () => {
    render(
      <Button
        onClick={() => console.log("clicked")}
        ariaLabel={"Go to Cart"}
        variant={"icon"}
      >
        <FontAwesomeIcon icon={faShoppingCart} />
      </Button>
    );
    expect(screen.getByRole("button", { name: "Go to Cart" })).toHaveAttribute(
      "aria-label"
    );
  });

  it("It should render without crashing", () => {
    render(
      <Button
        onClick={() => console.log("clicked")}
        ariaLabel={"Go to Cart"}
        variant={"icon"}
      >
        <FontAwesomeIcon icon={faShoppingCart} />
      </Button>
    );
    expect(
      screen.getByRole("button", { name: "Go to Cart" })
    ).toMatchSnapshot();
  });
});

describe("primary or secondary button", (): void => {
  it("should render a large primary button if no variant or size is defined", () => {
    render(
      <Button onClick={() => console.log("clicked")}>Primary Button</Button>
    );
    expect(screen.getByRole("button", { name: "Primary Button" })).toHaveClass(
      "primary-large"
    );
  });
});
