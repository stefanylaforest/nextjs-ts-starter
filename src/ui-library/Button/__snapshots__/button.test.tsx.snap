// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Only Button It should render without crashing 1`] = `
<button
  aria-label="Go to Cart"
  class="icon"
  type="button"
>
  <svg
    class="h-6 w-6"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18 18 6M6 6l12 12"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</button>
`;

exports[`primary or secondary button It should render without crashing 1`] = `
<button
  class="secondary large"
  disabled=""
  type="button"
>
  Secondary Button
</button>
`;
